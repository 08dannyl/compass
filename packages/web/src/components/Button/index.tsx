import { Styled, StyledFeedbackBtnContainer } from "./styled";

export const Button = Styled;

export const FeedbackButtonContainer = StyledFeedbackBtnContainer;
