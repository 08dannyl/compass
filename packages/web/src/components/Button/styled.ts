import styled from "styled-components";
import { ColorNames, InvertedColorNames } from "@web/common/types/styles";
import {
  getColor,
  getDarkerColor,
  getInvertedColor,
} from "@web/common/utils/colors";
import { Flex } from "@web/components/Flex";

export interface Props {
  color?: ColorNames;
  bordered?: boolean;
  border?: string;
}

export const StyledFeedbackBtnContainer = styled(Flex)`
  position: absolute;
  top: 10%;
  right: 8%;
`;

export const Styled = styled.div<Props>`
  background: ${({ color }) => getColor(color || ColorNames.BLUE_3)};
  min-width: 158px;
  padding: 0 8px;
  height: 36px;
  display: flex;
  align-items: center;
  font-weight: 600;
  justify-content: center;
  color: ${({ color = InvertedColorNames.BLUE_3 }) =>
    getInvertedColor(color as InvertedColorNames)};
  cursor: pointer;
  border-radius: 7px;
  border: ${({ bordered, color = InvertedColorNames.BLUE_3, border }) =>
    border ||
    (bordered && `2px solid ${getInvertedColor(color as InvertedColorNames)}`)};

  &:focus {
    border-width: ${({ bordered }) => (bordered ? 3 : 2)}px;
  }

  &:hover {
    background: ${({ color }) => getDarkerColor(color || ColorNames.BLUE_3)};
    transition: background-color 0.2s;
  }
`;
