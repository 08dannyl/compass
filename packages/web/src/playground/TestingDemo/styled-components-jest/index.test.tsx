/**
 * @jest-environment jsdom
 */
import React from "react";
import Button from ".";
import { render } from "@testing-library/react";
import "jest-styled-components";

describe("Testing Styles", () => {
  test("button snapshot works", () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });

  // test("color matches", () => {
  //   render(<Button />);
  //   expect(Button).toHaveStyleRule("color", "red");
  // });
});
