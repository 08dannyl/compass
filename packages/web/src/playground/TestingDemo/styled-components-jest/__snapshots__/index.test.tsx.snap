// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Styles button snapshot works 1`] = `
.c0 {
  color: red;
}

<button
  class="c0"
/>
`;
